Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -39,14 +34,39 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
 
             internal static void MoveInputManagerAssetActionsToProjectWideInputActionAsset()
             {
-                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
+                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPathInputManager);
                 if (objects != null)
                 {
                     var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
-                    if (inputActionsAsset != null)
+                    if (inputActionsAsset != default)
                     {
-                        var json = JsonUtility.ToJson(inputActionsAsset, prettyPrint: true);
-                        File.WriteAllText(ProjectWideActionsAsset.kDefaultAssetPath, json);
+                        // Found some actions in the InputManager.asset file
+                        //
+                        string path = ProjectWideActionsAsset.kDefaultAssetPath;
+
+                        if (File.Exists(path))
+                        {
+                            // We already have a path containing inputactions, find a new unique filename
+                            //
+                            //  eg  Assets/InputSystem_Actions.inputactions ->
+                            //      Assets/InputSystem_Actions (1).inputactions ->
+                            //      Assets/InputSystem_Actions (2).inputactions ...
+                            //
+                            string[] files = Directory.GetFiles("Assets", "*.inputactions");
+                            List<string> names = new List<string>();
+                            for (int i = 0; i < files.Length; i++)
+                            {
+                                names.Add(System.IO.Path.GetFileNameWithoutExtension(files[i]));
+                            }
+                            string unique = ObjectNames.GetUniqueName(names.ToArray(), "InputSystem_Actions");
+                            path = "Assets/" + unique + ".inputactions";
+                        }
+
+                        var json = inputActionsAsset.ToJson();
+                        File.WriteAllText(path, json);
```

Comment:
I would suggest removing this File.WriteAllText + AssetDatabase.Refresh() with a call to InputActionAssetManager.SaveAsset instead. In addition that method handles any underlying single-checkout versioning system which could otherwise generate a File-access exception here.

Reply:
Will make this change..